### DISCUSSION TITLE
Cannot access context in solid-ssr

### DISCUSSION THREAD

**Discussion by nhattran998** [vikejs/vike#748] [Discussion] [answered]
### Description

Because we do not have the example of solid-ssr with pageContext, I try this with the same concept of React, use Provider to wrap the page, and get value on `+onRenderClient` and `+onRenderHtml`. 

But in the home page component, when I use `usePageContext` --> Alway get the `undefined` value(You can see my log in the code). 

I hope you'll be able to take a look. When all things well, I might be can contribute the solid-ssr example with V1 design.

Reproduction: 
https://stackblitz.com/edit/solidjs-templates-awj35m?file=pages/index/+Page.tsx
https://github.com/harrytran998/vite-ssr-solid-v1

**Comment by brillout** [comment]
There is actually an [example of doing that](https://github.com/magne4000/solide/blob/5d91cfc903f02696ef2e80f3985a0ce9b44bf028/solide/renderer/PageContextProvider.tsx) &mdash; let me know whether that helps you resolved your issue.

>  I might be can contribute the solid-ssr example with V1 design.

Contributions, both to `examples/solid-*` and/or [Solide](https://github.com/magne4000/solide), welcome 💚.

**Reply by nhattran998** [reply]
Hi @brillout, I read the source code of Solide, I tried the same, but `usePageContent` still returns an `undefined` value. The provider already received the `pageContent` value, but `useContext` did not. 

If you have some time, please take a look at this repo. I already update the code like `solide`

https://stackblitz.com/edit/solidjs-templates-awj35m?file=composables/usePageContext.tsx

**Reply by brillout** [reply]
I'm probably the wrong person to ask as I'm not all too familiar with Solid.

But since Solide is a working example I guess it shouldn't be too difficult to find the problem?

**Comment by magne4000** [comment, marked as answer]
@harrytran998 :wave: I'm the one who wrote Solide, did you find where your issue came from?

**Reply by nhattran998** [reply]
@magne4000 Hi, thank you so much bcs solide, I tried and succeeded, and created an example with solid-ssr v1. I hope you can do some extraordinary things with solid + VPS 😎.
